applications/rag/README.md

```diff
@@ -1,192 +1,210 @@
+
+# Configure authenticated access via IAP (recommended)
+
+We recommend you configure authenticated access via IAP for your services.
+
+1) Make sure the [OAuth Consent Screen](https://developers.google.com/workspace/guides/configure-oauth-consent#configure_oauth_consent) is configured for your project. Ensure `User type` is set to `Internal`.
+2) Set the following variables in `workloads.tfvars`:
+    * `jupyter_add_auth = true`
+    * `frontend_add_auth = true`
+    * `ray_dashboard_add_auth = true`
+3) Allowlist principals for your services via `jupyter_members_allowlist`, `frontend_members_allowlist` and `ray_dashboard_members_allowlist`.
+4) Configure custom domains names via `jupyter_domain`, `frontend_domain` and `ray_dashboard_domain` for your services. If left blank, we'll provision test `nip.io` domains for you.
+5) Configure DNS records for your custom domains:
+    - [Register a Domain on Google Cloud Domains](https://cloud.google.com/domains/docs/register-domain#registering-a-domain) or use a domain registrar of your choice. You can skip this if using `nip.io`.
+    - Set up your DNS service to point to the public IP
+        * Run `terraform output frontend_ip_address` to get the public ip address of frontend, and add an A record in your DNS configuration to point to the public IP address.
+        * Run `terraform output jupyterhub_ip_address` to get the public ip address of jupyterhub, and add an A record in your DNS configuration to point to the public IP address.
+        * Run `terraform output ray_dashboard_ip_address` to get the public ip address of ray dashboard, and add an A record in your DNS configuration to point to the public IP address.
+    - Add an A record: If the DNS service of your domain is managed by [Google Cloud DNS managed zone](https://cloud.google.com/dns/docs/zones), there are two options to add the A record:
+        1. Go to https://console.cloud.google.com/net-services/dns/zones, select the zone and click ADD STANDARD, fill in your domain name and public IP address.
+        2. Run `gcloud dns record-sets create <domain address>. --zone=<zone name> --type="A" --ttl=<ttl in seconds> --rrdatas="<public ip address>"`
+
+# Cleanup
 
-    This filter can auto fetch the templates in your project. Please refer to the following links to create templates:
+1. Run `terraform destroy --var-file="workloads.tfvars"`
+    - Network deletion issue: `terraform destroy` fails to delete the network due to a known issue in the GCP provider. For now, the workaround is to manually delete it.
 
-    1. Inspect templates: https://cloud.google.com/dlp/docs/creating-templates-inspect
-    2. De-identification templates: https://cloud.google.com/dlp/docs/creating-templates-deid
+# Troubleshooting
 
-## Verify setup
+Set the namespace (change to the namespace in `workloads.tfvars` if using a custom `kubernetes_namespace`):
```

Comment:
Likewise, the previous wording is more clear. Perhaps add a link to the Installation section?

`Set your namespace from workloads.tfvars:`